hamaddr/src/error.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use std::fmt::{Debug, Display, Formatter};
+
+pub type Result<T = (), E = anyhow::Error> = std::result::Result<T, E>;
+
+/// Error type indicating an invalid character at a specific index.
+#[derive(Debug, thiserror::Error)]
+pub struct InvalidCharAt(pub usize);
+
+impl Display for InvalidCharAt {
```

Comment:
Q: can you derive Display instead of using an explicit impl?

Reply:
Nope, `Display` can't be derrived.

Reply:
Ah, I was thinking of derive_more::Display.